### PR TITLE
screenconstraint more robust against nans and flags

### PR DESCRIPTION
**Pull request by maaijke**

### CODE REVIEW

File: DPPP/SolTab.cc [hunk 1]
```diff
@@ -323,7 +323,7 @@ namespace DP3 {
     try {
       val.read(res.data(), H5::PredType::NATIVE_DOUBLE, memspace, dataspace);
     } catch (H5::DataSetIException& e) {
-      e.printErrorStack();
+      e.printError();
```
Comment (tammojan) [inline DPPP/SolTab.cc]: Please do not check this in, it's unrelated to the change you're making.

File: DDECal/KLFitter.cc [hunk 2]
```diff
@@ -59,6 +59,7 @@ void KLFitter::calculateCorrMatrix(const std::vector<PiercePoint*> pp){
 
 
 void KLFitter::doFit(){
+
```
Comment (tammojan) [inline DDECal/KLFitter.cc]: This newline does not really add a lot to the code :)

File: DDECal/ScreenConstraint.cc [hunk 3]
```diff
@@ -157,32 +157,56 @@ void ScreenConstraint::CalculatePiercepoints(){
   void  ScreenConstraint::getPPValue(std::vector<std::vector<MultiDirSolver::DComplex> >& solutions,size_t solutionIndex,size_t dirIndex,double &avgTEC,double &error) const {
   if (itsAVGMode=="simple"){
     avgTEC=0;
+    error=1.;
+    uint nrch(0);
+    for(size_t ch=0;ch<_nChannelBlocks; ++ch){
+      if(isfinite(solutions[ch][dirIndex])){
+	double refphase=std::arg(solutions[ch][dirIndex]);
+	//TODO: more advance frequency averaging...
```
Comment (tammojan) [inline DDECal/ScreenConstraint.cc]: Please indent with a multiple of two spaces (and no tabs)

File: DPPP/PhaseFitter.cc [hunk 4]
```diff
@@ -24,6 +24,7 @@
 #include "PhaseFitter.h"
 
 #include <limits>
+#include <iostream>
```
Comment (tammojan) [inline DPPP/PhaseFitter.cc]: can be removed

File: DPPP/PhaseFitter.cc [hunk 5]
```diff
@@ -48,7 +49,7 @@ double PhaseFitter::fitTEC2ModelBeta(double alpha, double betaEstimate) const {
   for(size_t iter=0; iter!=3; ++iter) {
     double sum = 0.0;
     for(size_t i=0; i!=Size(); ++i) {
-      double p = _phases[i], e = TEC2ModelFunc(_frequencies[i], alpha, betaEstimate);
+       double p = _phases[i], e = TEC2ModelFunc(_frequencies[i], alpha, betaEstimate);
```
Comment (tammojan) [inline DPPP/PhaseFitter.cc]: This space should not be here.

File: DPPP/PhaseFitter.h [hunk 6]
```diff
@@ -264,7 +265,7 @@ class PhaseFitter
 	 */
   static double TEC2ModelFuncWrapped(double nu, double alpha, double beta)
   {
-    return fmod(alpha / nu + beta, 2.0*M_PI);
+     return fmod(alpha / nu + beta, 2.0*M_PI);
```
Comment (tammojan) [inline DPPP/PhaseFitter.h]: This space should not be here.

File: DDECal/ScreenConstraint.cc [hunk 7]
```diff
@@ -157,32 +157,56 @@ void ScreenConstraint::CalculatePiercepoints(){
...
+	if (isfinite(solutions[ch][solutionIndex])){
+	avgTEC += std::arg(solutions[ch][solutionIndex]*std::polar<double>(1.0,-1*refphase))*itsFrequencies[ch]*phtoTEC;
+	nrch++;
+	  }
+      }
+    }
+    if (nrch>0)
+      avgTEC/=nrch;
+    /*
+    double mydelay=0;
```
Comment (tammojan) [inline DDECal/ScreenConstraint.cc]: Why is this code commented out?

Comment (aroffringa) [reply DDECal/ScreenConstraint.cc:175]: It's new code, so I'm just accepting it as is for now

**Pull request merged**